VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseDiscussionResponsesAdapter.java

```diff
@@ -390,12 +391,24 @@ public void addAll(List<DiscussionComment> items) {
         notifyDataSetChanged();
     }
 
-    public void setDiscussionThread(@NonNull DiscussionThread discussionThread) {
-        this.discussionThread = discussionThread;
-        this.discussionResponses.clear();
+    public void addNewResponse(@NonNull DiscussionComment response) {
+        discussionResponses.add(response);
         notifyDataSetChanged();
     }
 
+    public void addNewComment(@NonNull DiscussionComment parent) {
+        String parentId = parent.getIdentifier();
+        if (!TextUtils.isEmpty(parentId)) {
```

Comment:
I don't think we need to validate the identifier at this point. Any validation should be performed once at the time of deserialization.